server/services/visitSessionsService.ts

```diff
@@ -158,11 +158,13 @@ export default class VisitSessionsService {
           const timeEnd = parse(`${visitSlotsforDay.date} ${year} 23:59`, 'EEEE d MMMM yyyy H:mm', new Date())
           visitSlotsforDay.prisonerEvents.afternoon = getPrisonerEvents(prisonerEvents, timeStart, timeEnd)
         }
-
-        allVisitSlots.push(visitSlotsforDay)
+        if (visitSlotsforDay.slots.morning.length > 0 || visitSlotsforDay.slots.afternoon.length > 0) {
```

Comment:
I'm not sure this is in the right place, it's essentially saying if there are no sessions for this day, remove the month. You can see the `availableSessions[month] = allVisitSlots` has moved inside the inner loop. The `if` and the following line are correct however the other 2 lines need to happen after the `availableSessions[month].forEach` loop with a different condition on the `if`. So the `if` would maybe be `if (availableSessions[month].length`?

The `if` for the `visitSlotsforDay` is correct, but the one for `availableSessions[month]` should be after the `availableSessions[month]` loop.

Reply:
re-implemented this outside of the month's loop in my most recent commit